src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -9090,11 +9075,14 @@ namespace WaterThermalTanks {
             } else if (WaterHeaterDesuperheater(DesuperheaterNum).ReclaimHeatingSource == CONDENSER_REFRIGERATION) {
                 HeatReclaimRefrigCondenser(SourceID).UsedWaterHeater = WaterHeaterDesuperheater(DesuperheaterNum).HeaterRate;
             } else if (WaterHeaterDesuperheater(DesuperheaterNum).ReclaimHeatingSource == COIL_DX_COOLING ||
-                       WaterHeaterDesuperheater(DesuperheaterNum).ReclaimHeatingSource == COIL_DX_MULTISPEED ||
-                       WaterHeaterDesuperheater(DesuperheaterNum).ReclaimHeatingSource == COIL_DX_MULTIMODE) {
+                WaterHeaterDesuperheater(DesuperheaterNum).ReclaimHeatingSource == COIL_DX_MULTISPEED ||
+                WaterHeaterDesuperheater(DesuperheaterNum).ReclaimHeatingSource == COIL_DX_MULTIMODE) {
                 HeatReclaimDXCoil(SourceID).AvailCapacity -= WaterHeaterDesuperheater(DesuperheaterNum).HeaterRate;
             } else if (WaterHeaterDesuperheater(DesuperheaterNum).ReclaimHeatingSource == COIL_DX_VARIABLE_COOLING) {
                 DataHeatBalance::HeatReclaimVS_DXCoil(SourceID).AvailCapacity -= WaterHeaterDesuperheater(DesuperheaterNum).HeaterRate;
+            } else if (WaterHeaterDesuperheater(DesuperheaterNum).ReclaimHeatingSource == COIL_AIR_WATER_HEATPUMP_EQ) {
+                DataHeatBalance::HeatReclaimSimple_WAHPCoil(SourceID).AvailCapacity -= WaterHeaterDesuperheater(DesuperheaterNum).HeaterRate; 
+                DataHeatBalance::HeatReclaimSimple_WAHPCoil(SourceID).WaterHeatingDesuperheaterReclaimedHeat(DesuperheaterNum) = WaterHeaterDesuperheater(DesuperheaterNum).HeaterRate;
```

Comment:
This might work just fine. Maybe I was being too paranoid. It looks like this bucket is specific to Coil:WaterHeating:Desuperheater. However, it's always good to test with multiple objects (e.g., both air and water desuperheater types) in the input file because we find problems later when not tested thoroughly enough. So is this the only bucket that changed to an array? and the others are still a sum?

Reply:
The bottom line is @yzhou601 has done enough here and we all learned a lot along the way. No other changes should be necessary for now since it appears to work as intended. If we think it would help if other buckets need to be arrays then that can be addressed in a follow up issue as a suggestion.

Reply:
Thanks @rraustad , I agree that the array would only account for Coil:WaterHeating:Desuperheater object, but since the Coil:Heating:Desuperheater is not able to model source from water coils currently. they are intended to be tracked in different structs. My thought is to add other user capabilities as separate arrays in that struct if they're available in the future. 

And thanks very much for your help along the way, very glad to receive so many very useful feedback and suggestions! Btw, there's another coil capability (DX multispeed) sitting in another branch, which is not as much work as this one. I would merge that one into this branch to make sure everything is ready.